### DISCUSSION TITLE
jotai contains an invalid package.json configuration

### DISCUSSION THREAD

**Discussion by toni1703** [pmndrs/jotai#2096] [Bug report]
jotai package has an incorrect package.json configuration. Specifically, the exports field in the package.json is pointing to .mjs files that do not exist within the package's directory structure.

warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for " /Users/bob/Documents/project/apps/mobile/node_modules/jotai" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/index.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for " /Users/bob/Documents/project/apps/mobile/node_modules/jotai/vanilla" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/vanilla.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/react" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/react.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/index.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/index.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/index.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/utils" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/utils.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/vanilla/utils" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/vanilla/utils.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/react/utils" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/react/utils.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/react" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/react.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/vanilla/utils" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/vanilla/utils.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/vanilla" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/vanilla.mjs, however this file does not exist. Falling back to file-based resolution.
warning: The package /Users/bob/Documents/project/apps/mobile/node_modules/jotai contains an invalid package.json configuration. Consider raising this issue with the package maintainer(s).
Reason: The resolution for "/Users/bob/Documents/project/apps/mobile/node_modules/jotai/vanilla" defined in "exports" is /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/vanilla.mjs, however this file does not exist. Falling back to file-based resolution.

I'm in an nx monorepo expo managed react native app.

**Comment by dai-shi** [comment]
Hi, can you list files in `/Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/` ?

It can probably be nx specific issue. Might be related with https://github.com/pmndrs/zustand/discussions/1930. cc @barelyhuman

**Reply by barelyhuman** [reply]
@toni1703 can you help us a bit more with the following information as well?
- Jotai version you are using 
- Nx version being used 
- also, is the package installed on the root of the project or in the same project or is it being required by a sibling project

If it's 2.4.0 
You can validate that these files exist in the version published here https://unpkg.com/browse/jotai@2.4.0/esm/

So, we'll need a few more details to proceed

**Comment by toni1703** [comment]
On Sunday, August 27, 2023 at 12:22:46 PM GMT+3, Daishi Kato ***@***.***> wrote:  
 
 


Hi, can you list files in /Users/bob/Documents/project/apps/mobile/node_modules/jotai/esm/ ?

It can probably be nx specific issue. Might be related with pmndrs/zustand#1930. cc @barelyhuman

**Comment by toni1703** [comment]
Jotai v:2.4.0Nx v:16.7.4So i have a Nx expo managed monorepo. the jotai is installed in root. in my apps i have a project called "mobile" with a package.json that need jotai, but jotai comes from root. ("jotai": "*",)    On Sunday, August 27, 2023 at 03:52:28 PM GMT+3, reaper ***@***.***> wrote:  
 
 


@toni1703 can you help us a bit more with the following information as well?
   
   - Jotai version you are using
   - Nx version being used
   - also, is the package installed on the root of the project or in the same project or is it being required by a sibling project

If it's 2.4.0
You can validate that these files exist in the version published here ***@***.***/esm/

So, we'll need a few more details to proceed

**Comment by toni1703** [comment]
I've checked for the vanilla.mjs and there are not placed in my node modules    On Sunday, August 27, 2023 at 04:15:38 PM GMT+3, Ioanas Toni ***@***.***> wrote:  
 
  Jotai v:2.4.0Nx v:16.7.4So i have a Nx expo managed monorepo. the jotai is installed in root. in my apps i have a project called "mobile" with a package.json that need jotai, but jotai comes from root. ("jotai": "*",)    On Sunday, August 27, 2023 at 03:52:28 PM GMT+3, reaper ***@***.***> wrote:  
 
 


@toni1703 can you help us a bit more with the following information as well?
   
   - Jotai version you are using
   - Nx version being used
   - also, is the package installed on the root of the project or in the same project or is it being required by a sibling project

If it's 2.4.0
You can validate that these files exist in the version published here ***@***.***/esm/

So, we'll need a few more details to proceed

**Reply by barelyhuman** [reply]
Okay, I’m that case the linker that nx sets up with react native’s bundler (metro) might not  be working as needed and you’ve mentioned that the files don’t exist. 

For problem 2. Can you just get rid of all the node_modules folders and run an install again? 

If it starts showing the files this time, and still throw an error then we can move towards problem 1

**Comment by toni1703** [comment]
yes after i deleted the node_modules, and run yarn, the files can't be found
    On Sunday, August 27, 2023 at 08:09:49 PM GMT+3, reaper ***@***.***> wrote:  
 
 


Okay, I’m that case the linker that nx sets up with react native’s bundler (metro) might not be working as needed and you’ve mentioned that the files don’t exist.

For problem 2. Can you just get rid of all the node_modules folders and run an install again?

If it starts showing the files this time, and still throw an error then we can move towards problem 1

**Reply by barelyhuman** [reply]
Sorry for the delay, we’ll need a reproduction repo at this point, it’s hard to solve without knowing what part of your tooling is causing it
